src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -243,6 +264,31 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
         }
     }
 
+    private int[] findTopApproximatePostings(int limit, int nRows, IntStream segmentRowIdsStream, NodeSimilarity.ApproximateScoreFunction scoreFunction) throws IOException
+    {
+        ArrayList<Pair<Integer, Float>> pairs = new ArrayList<>(nRows);
+        try (var ordinalsView = graph.getOrdinalsView())
+        {
+            // just because getOrdinalForRowId throw IOException, to avoid wrapping it into RuntimException
+            Iterable<Integer> segmentRowIdIterator = segmentRowIdsStream::iterator;
```

Comment:
this will cause box/unbox unnecessarily, use PrimitiveIterator.ofInt to avoid this